### PR TITLE
Performance improvements for the JSON transformation SMT

### PR DESCRIPTION
**Pull request by sroze**
In short, we go from ~2200 ops/secs to ~3500 ops/sec, with the schema union shenanigans 🎉

### CODE REVIEW

File: src/main/java/com/birdie/kafka/connect/json/SchemaTransformer.java [hunk 1]
```diff
@@ -191,56 +205,59 @@ public SchemaBuilder unionSchemas(Schema ...schemas) {
...
+
+            if (!schema.type().equals(type)) {
+                throw new IllegalArgumentException("We can only union schemas of the same type together. Found: " + type + " and " + schema.type());
+            }
+
+            if (schema.type().equals(Schema.Type.ARRAY)) {
+                if (valueType == null) {
+                    valueType = schema.valueSchema().type();
+                } else if (!valueType.equals(schema.valueSchema().type())) {
+                    throw new IllegalArgumentException("We can only union array schemas of the same value type together. Found: " + valueType + " and "+ schema.valueSchema().type());
```
Comment (airwalk225) [inline src/main/java/com/birdie/kafka/connect/json/SchemaTransformer.java:228]: Do we expect to hit these use cases much?

**Review by airwalk225 — APPROVED** [review]

File: src/main/java/com/birdie/kafka/connect/json/SchemaTransformer.java (hunk 1 above)
Comment (sroze) [reply src/main/java/com/birdie/kafka/connect/json/SchemaTransformer.java:228]: Yup yup it happens 😅

Comment (airwalk225) [reply src/main/java/com/birdie/kafka/connect/json/SchemaTransformer.java:228]: The assumption being that we _do_ create a new schema at this point? How do we keep track of that, or do we just let it fail?

Comment (sroze) [reply src/main/java/com/birdie/kafka/connect/json/SchemaTransformer.java:228]: Yes exactly, there is another part of the code which creates an alternative schema. In practices, there is a list of schemas it tries to unify with, not just one.

**Review by ChloePont — APPROVED** [review]
🏎️

**Pull request merged**